include/clap/factory/draft/preset-discovery.h

```diff
@@ -0,0 +1,294 @@
+
+   // This location/collection is meant for storing user created presets
+   CLAP_PRESET_DISCOVERY_IS_USER_CONTENT = 1 << 1,
+
+   // This location/collection is meant for demo presets, those are preset which may trigger
+   // some limitation in the plugin because they require additionnal features which the user
+   // needs to purchase or the content itself needs to be bought and is only available in
+   // demo mode.
+   CLAP_PRESET_DISCOVERY_IS_DEMO_CONTENT = 1 << 2,
+
+   // This preset is a user's favorite
+   CLAP_PRESET_DISCOVERY_IS_FAVORITE = 1 << 3,
+};
+
+// Receiver that receives the metadata for a single preset file.
+// The host would define the various callbacks in this interface and the preset parser function
+// would then call them.
+//
+// This interface isn't thread-safe.
+typedef const struct clap_preset_discovery_metadata_receiver {
+   // If there is an error reading metadata from a file this should be called with an error
+   // message.
+   // os_error: the operating system error, if applicable. If not applicable set it to a non-error
+   // value: 0 on unix and Windows.
+   void(CLAP_ABI *on_error)(const struct clap_preset_discovery_metadata_receiver *receiver,
+                            int32_t                                               os_error,
+                            const char                                           *error_message);
+
+   // Marks this file as a container file meaning that it can contain other presets.
+   void(CLAP_ABI *mark_as_container_file)(
+      const struct clap_preset_discovery_metadata_receiver *receiver);
+
+   // If the file being parsed is a preset container file (mark_as_container_file has been called)
+   // then this must be called for every preset in the file and before any preset metadata is
+   // sent with the calls below. If the file is not a container file then this should not be
+   // called at all.
+   //
+   // The path defines a human friendly path to the preset in the container file. It
+   // should be unique within the container file.
+   //
+   // The preset_id is a machine friendly string used to load the preset inside the container via a
+   // the preset-load plug-in extension. The preset_id can also just be the path if that's what the
+   // extension wants but it could also be some other unique id like a database primary key or a
+   // binary offset. It's use is entirely up to the plug-in.
+   void(CLAP_ABI *begin_contained_preset)(
+      const struct clap_preset_discovery_metadata_receiver *receiver,
+      const char                                           *path,
+      const char                                           *preset_id);
+
+   // Marks this preset as a bank preset, meaning that it can be assigned to the plug-in as a
+   // preset but will update the banks in the plug-in.
+   void(CLAP_ABI *mark_as_bank_preset)(
+      const struct clap_preset_discovery_metadata_receiver *receiver);
```

Comment:
Perhaps this can be replaced by a clap_preset_discovery_flags flag?

Reply:
Yes it can!